pom.xml

```diff
@@ -522,6 +523,17 @@
                     </sourcepath>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven-compiler-plugin.version}</version>
+                <configuration>
+                    <source>11</source>
+                    <target>11</target>
+                    <compilerVersion>11</compilerVersion>
```

Comment:
We don't need these, they are defined in the `properties`, is removing these ok in your case?

```suggestion
```

Reply:
If we use `maven.compiler.source` he will read the jdk version of the maven setting, if we want to define variables we need to define different variable names

Reply:
@weihubeats read this https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#source and https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#target

> <source>
The -source argument for the Java compiler.
Type: java.lang.String
Required: No
User Property: maven.compiler.source // <<< 👈 this one
Default: 1.8


the equivalent property is `maven.compiler.source` and we already defined it in the `<properties>` section, so I mean we don't need to set `<source>/<target>/<compilerVersion>` here

Reply:
`maven.compiler.source` doesn't work, we need to define

Reply:
My test defaults to reading the jdk version in maven settings

Reply:
At least using `maven.compiler.source` and `maven.compiler.target`, rather than hard coded.

Reply:
> `maven.compiler.source` doesn't work, we need to define

This doesn't make sense to me, it's clearly documented in maven's documentation, the property `maven.compiler.source` is equivalent to the `<source>` in `<configuration`

Reply:
We need to verify why it doesn't work before apply any change.

Reply:
I know, but if we use the `maven.compiler.source` variable name he will still read the jdk version of the maven setting, so we need to define a new variable

Reply:
I am afraid we need to follow doc comes first, especially it works on others' env.

Reply:
We change the variable name `maven.compiler.source` to `skywalking.maven.compiler.source`?

Reply:
We need why for any change

Reply:
Because using `${maven.compiler.source}` he will not read the version defined in the project, but will still read the jdk version in maven setttings.xml, so we need to define a different variable

Reply:
Or define a new variable for `maven-compiler-plugin` to use

Reply:
> I know, but if we use the `maven.compiler.source` variable name he will still read the jdk version of the maven setting, so we need to define a new variable

Then it would be a bug of JetBrains. If so, you should raise a ticket in their support system rather than manually fix configurations in SkyWalking.